Mathlib/RingTheory/Kaehler.lean

```diff
@@ -388,18 +385,42 @@ theorem KaehlerDifferential.End_equiv_aux (f : S →ₐ[R] S ⊗ S ⧸ KaehlerDi
     exact e₁.symm.trans (e.trans e₂)
 #align kaehler_differential.End_equiv_aux KaehlerDifferential.End_equiv_aux
 
-set_option maxHeartbeats 700000 in
--- Porting note: extra heartbeats are needed to infer the instance
--- Module S { x // x ∈ Ideal.cotangentIdeal (ideal R S) }
-set_option synthInstance.maxHeartbeats 200000 in
--- This has type
--- `Derivation R S (Ω[S⁄R]) ≃ₗ[R] Derivation R S (KaehlerDifferential.ideal R S).cotangentIdeal`
--- But lean times-out if this is given explicitly.
+/- Note: Lean is slow to synthesize theses instances (times out).
+  Without them the endEquivDerivation' and endEquivAuxEquiv both have significant timeouts.
+  In Mathlib 3, it was slow but not this slow. -/
+local instance : SMul (S ⊗[R] S ⧸ KaehlerDifferential.ideal R S ^ 2)
+    (S ⊗[R] S ⧸ KaehlerDifferential.ideal R S ^ 2) := Mul.toSMul _
+
+theorem isScalarTower_S_right : IsScalarTower S (S ⊗[R] S ⧸ KaehlerDifferential.ideal R S ^ 2)
+    (S ⊗[R] S ⧸ KaehlerDifferential.ideal R S ^ 2) := IsScalarTower.right
+
+attribute [local instance] isScalarTower_S_right
```

Comment:
Why a `theorem` and not `local instance`?

Reply:
`defLemma` complained. Is there a better fix?

Reply:
That sounds like a bug in `local instance` that we should report somewhere. I think a minimal repro is
```lean
local instance : Fact True := sorry
```

Reply:
It isn't making that easy for me. But `local` is definitely the trigger. I am going to try to minimize in a bit. This has occurred for me before with `Fact` deep into the library. Maybe the import is messing things up? We'll see.

Reply:
I think the problem is that `instance` is just always a `def`

Reply:
It is a `def` with or without the local. Should [this](https://github.com/leanprover/std4/blob/e3c2be331da9ddeef7f82ca363f072a68d7210b3/Std/Tactic/Lint/Misc.lean#L96)
```lean 
    if (← isAutoDecl declName) || isGlobalInstance (← getEnv) declName then
```
be 
```lean
    if (← isAutoDecl declName) || (← isInstance declName) then
```
?

Reply:
Hmm. Probably out of scope regardless

Reply:
I didn't figure out how to read local instances in the scope that `#lint` works in. For now, it seems better just to `nolint` and bring the issue up on [Zulip](https://leanprover.zulipchat.com/#narrow/stream/348111-std4/topic/local.20instances.20in.20.23lint).